go/tools/builders/cgo2.go

```diff
@@ -95,7 +98,27 @@ func cgo2(goenv *env, goSrcs, cgoSrcs, cSrcs, cxxSrcs, objcSrcs, objcxxSrcs, sSr
 		}
 	}
 	combinedLdFlags = append(combinedLdFlags, defaultLdFlags()...)
-	os.Setenv("CGO_LDFLAGS", strings.Join(combinedLdFlags, " "))
+
+	var ldflagsFile *os.File
+	combinedLdFlagsStr := strings.Join(combinedLdFlags, " ")
+	if len(combinedLdFlagsStr) < maxEnvVarSize {
+		os.Setenv("CGO_LDFLAGS", combinedLdFlagsStr)
+	} else {
+		// Write linker flags to a temporary file instead of pasing via env variable.
+		// This avoids "argument list too long" error with extremely large CGO_LDFLAGS
+		// that can exceed system limits.
+		ldflagsFile, err = os.Create(filepath.Join(workDir, "cgo-ldflags.txt"))
```

Comment:
Create a temporary file for this with `os.CreateTemp`. Make sure to delete it with a `defer` call.